### DISCUSSION TITLE
birt > 4.7

### DISCUSSION THREAD

**Discussion by NoLamprecht** [eclipse-birt/birt#1009] [General]
going java > 8
all itext from 5. to 7.
test birt 4.8 4.9 4.10
load birt source and debug
problem is in  ExtensionManager -> no pdf emitter in any table
i go over ID - i go "pdf" i go over both -> error no pdf support !
in osgi runtime ExtensionManager NO pdf extension found
after loadEmitterExtensionDefns(); no pdf support found
also -> error no pdf support !

**Comment by wimjongman** [comment]
Please provide detailed instructions on how to duplicate this problem. Your explanation may make sense to you but for me, it is completely unclear what the problem is.

**Comment by NoLamprecht** [comment]
the pdf emitter is not loaded
i write a mini exampel

**Comment by NoLamprecht** [comment]
https://github.com/NoLamprecht/BWPrint

show the doc -> many jars not modular

**Comment by NoLamprecht** [comment]
JAVA 11 -17 -> Zulu JDKFX im Test

**Comment by hvbtup** [comment]
You wrote: "all itext from 5. to 7.".
BIRT did never use iText 5 and does not use iText 7.
Older BIRT used iText 2.1.7.
We switched from iText to OpenPDF with BIRT 4.9.

**Comment by NoLamprecht** [comment]
My target in my programm
JAVA 17 FX
ITEXT7 
and all modular

thats only my Background and why i need Birt 4.9

**Comment by hvbtup** [comment]
BIRT does not support iText 7.

**Comment by NoLamprecht** [comment]
MY USE IS iText7 --> NOT BIRT -> but you kown my libs in use

look the problem file please -> there are very old jars !! not modular !!

the report Lib files NOT to use !! so i think the osgi loading not ok

THE Birt is NOT integrating to java > 8 in real programms 
You use FX and Birt libs not to  use

**Comment by hvbtup** [comment]
Please try to be polite and use whole sentences if you want anyone to help you.

Note: Personally I only ever use the POJO version with "classic" class loading and can't tell much about the OSGI runtime or newer class loading mechanisms.  Please take this into account when reading my comments.

I briefly looked at the source code you provided. The code looks more or less ok at first sight.
However, I'm pretty sure that there are at least two mistakes:

First of all, your project is missing a great part of the BIRT runtime libs: The whole ReportEngine/lib is missing.

And inside Print.java, method aBirtEngine, it seems like you try to handle the POJO runtime and the OSGI runtime. AFAIK, BIRT_HOME should only be set for the OSGI version and might cause difficulties for the POJO version.

See the file ReportEngine/genReport.bat for an example how to setup the class path (the same script works for the OSGI and for the POJO runtime).

It seems to me that your issues are somewhat related to the class loading mechanism you are using.

I am using BIRT 4.9 with Java 11 in production code without any problems (with the POJO runtime and classic class loading). So your statement "BIRT is not integrating to java > 8 in real programs" is not true.

What might be true (I don't know) is that you are one of the first persons to try to use BIRT with Java 11 modularization.

We can only hope that someone else has already tried that and can help here.

**Comment by NoLamprecht** [comment]
so i say above,

First of all, your project is missing a great part of the BIRT runtime libs: The whole ReportEngine/lib is missing.

Use FX und you cannot use the Report Libs - to old - not modular
Thats my problem
